### DISCUSSION TITLE
Azure WebApp - Narzędzie dla helpdesku do restartowania webappki

### DISCUSSION THREAD

**Discussion by maciejfrydrych** [AzureCommunityPL/group#94] [Pomoc] [answered]
Cześć. :) Przychodzę z powyższym pytaniem. Nie jestem programistą, więc raczej nic sobie nic nie napiszę. Co najwyżej jakiś prosty skrypt. 

Mam różne webappki na jeden z subskrypcji. Jedna z nią jest strona, która co jakiś czas trzeba restartować. Potrzebuje zrobić mechanizm, który będzie pozwalał ludkom z Helpdesku, w łatwy sposób zrestartować tą stronę. 

Myślałem o tym, aby zrobić service principala, który będzie miał uprawnienia tylko do restartu webappki. Ale nie wiem czy jest możliwe przycięcie uprawnień, na takim poziomie.  Nie mam pomysłu jak im to udostępnić, aby nie było skomplikowane i tutaj liczę na jakaś podpowiedź.

**Comment by wi5nia** [comment, marked as answer]
Jak maja wjazd na portal Azure to mozesz im [custom role](https://docs.microsoft.com/en-us/azure/role-based-access-control/resource-provider-operations#web) zrobic
```
Microsoft.Web/sites/restart/Action | Restart a Web App
```
A po prawdzie to lepiej sie zajmij naprawa apki zeby nie trzeba bylo robic takich jaj :)

**Reply by komluk** [reply]
Dokładnie!
Zamiast kombinować nad tymczasowym rozwiązaniem z restartowaniem lepiej naprawić aplikacje tak jak sugeruje @wi5nia ;)

**Reply by aryll** [reply]
To już raczej kwestia zarządzania w firmie - a z tym jak wiemy jest różnie 😉

**Reply by kamil-mrzyglod** [reply]
No bez przesady - mógł to ktoś przekalkulować i wyszło na to, że jak ludek raz na jakiś czas wejdzie i kliknie restart to wyjdzie taniej niż szukanie dziury ;)

**Reply by kamil-mrzyglod** [reply]
BTW - jak im zrobisz custom rolę tak jak @wi5nia sugeruje to możesz im dać plik z następującą komendą do uruchomienia - https://docs.microsoft.com/en-us/cli/azure/webapp?view=azure-cli-latest#az_webapp_restart:
```
az webapp restart --name MyWebapp --resource-group MyResourceGroup
```
Nawet nie muszą wiedzieć, że jest jakiś portal 😉 Ewentualnie to samo tylko w Powershell.

**Reply by aryll** [reply]
@kamil-mrzyglod  ja się z Tobą zgadzam, ale za dużo takich rzeczy widziałam żeby mieć pewność że wszyscy potrafią kalkulować...

**Reply by maciejfrydrych** [reply]
> To już raczej kwestia zarządzania w firmie - a z tym jak wiemy jest różnie 😉

Dokładnie, nie moja appka, ja już to proponowałem, łącznie z ulepszeniem jest aby korzystala z [Self Healing](https://docs.microsoft.com/en-us/azure/architecture/guide/design-principles/self-healing). Ale jak widać prościej jest ;) jak ktoś ją będzie restartował.

**Reply by maciejfrydrych** [reply]
> BTW - jak im zrobisz custom rolę tak jak @wi5nia sugeruje to możesz im dać plik z następującą komendą do uruchomienia - https://docs.microsoft.com/en-us/cli/azure/webapp?view=azure-cli-latest#az_webapp_restart:
> 
> ```
> az webapp restart --name MyWebapp --resource-group MyResourceGroup
> ```
> 
> Nawet nie muszą wiedzieć, że jest jakiś portal 😉 Ewentualnie to samo tylko w Powershell.

Dzięki za podpowiedź. Razem z tą od @wi5nia coś wystrugam.

**Reply by wi5nia** [reply]
@kamil-mrzyglod jak to konto uzytkownika to badz czlowikiem i daj komus ladne kolorki w portalu zobaczyc, a nie pchasz w smutny terminal... ;) 
a jak konto serwisowe to owszem, automatyzacja zawsze
